pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -2100,3 +2123,74 @@ func getRouteHostname(obj interface{}) (host string) {
 	}
 	return host
 }
+
+func checkAndCreateAdminSecret(service kubernetes.PlatformService, namespace string, secretName string, environment api.EnvironmentType) (error, string, string) {
+	/* The default secret will be like this
+
+	apiVersion: v1
+	kind: Secret
+	metadata:
+	  name: kie-admin-credentials
+	type: Opaque
+	data:
+	  //adminUser
+	  username: YWRtaW4=
+	  //RedHat
+	  password: UmVkSGF0
+
+	*/
+
+	customerSecret, err := getSecret(service, namespace, secretName)
+
+	// The secret passed by the user doesn't exist we create one with default values
+	if err != nil {
+		username := constants.DefaultAdminUser
```

Comment:
probably it is a good idea remove the blank user and pasword from cr and cr.applied after the secret is in place.
@jakubschwan @desmax74 wdyt?

Reply:
Depends maybe someone is still using the "old" behaviour and is fine for they, now they could choose.